pkg/dynamicaccess/grantee.go

```diff
@@ -1,44 +1,139 @@
 package dynamicaccess
 
 import (
+	"context"
 	"crypto/ecdsa"
+	"crypto/elliptic"
+	"fmt"
+
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/storer"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const (
+	publicKeyLen = 65
 )
 
+// TODO: maybe rename to "List", simply
 type GranteeList interface {
-	Add(topic string, addList []*ecdsa.PublicKey) error
-	Remove(topic string, removeList []*ecdsa.PublicKey) error
-	Get(topic string) []*ecdsa.PublicKey
+	Add(publicKeys []*ecdsa.PublicKey) error
+	Remove(removeList []*ecdsa.PublicKey) error
+	Get() []*ecdsa.PublicKey
+	Save() (swarm.Address, error)
 }
 
 type GranteeListStruct struct {
-	grantees map[string][]*ecdsa.PublicKey
+	grantees []byte
+	loadSave file.LoadSaver
+	putter   storer.PutterSession
+}
+
+var _ GranteeList = (*GranteeListStruct)(nil)
+
+func (g *GranteeListStruct) Get() []*ecdsa.PublicKey {
+	return g.Deserialize(g.grantees)
+}
+
+func (g *GranteeListStruct) Serialize(publicKeys []*ecdsa.PublicKey) []byte {
+	b := make([]byte, 0, len(publicKeys))
```

Comment:
I think this probably should be `b := make([]byte, 0, len(publicKeys)*publicKeyLen)`